hypervisor/network.go

```diff
@@ -201,6 +235,64 @@ func (nc *NetworkContext) removeInterface(id string, result chan api.Result) {
 	}
 }
 
+// allInterfaces return all the network interfaces except loop
+func (nc *NetworkContext) allInterfaces() (nics []*InterfaceCreated) {
+	nc.slotLock.Lock()
+	defer nc.slotLock.Unlock()
+
+	for _, v := range nc.eth {
+		if v != nil {
+			nics = append(nics, v)
+		}
+	}
+	return
+}
+
+func (nc *NetworkContext) updateInterface(inf *api.InterfaceDescription) error {
+	oldInf, ok := nc.idMap[inf.Id]
+	if !ok {
+		nc.sandbox.Log(WARNING, "trying update a non-exist interface %s", inf.Id)
+		return fmt.Errorf("interface %q not exists", inf.Id)
+	}
+
+	// only support update some fields: Name, ip addresses, mtu
+	nc.slotLock.Lock()
+	defer nc.slotLock.Unlock()
+
+	if inf.Name != "" {
+		oldInf.NewName = inf.Name
```

Comment:
Is it necessary to add a new field NewName?Can not save new name to DeviceName?

Reply:
There's one trouble here, I add a `NewName` for modifying device name in Guest OS.

in struct `InterfaceCreated`, there are 3 fields in my codes:
```
HostDevice string
DeviceName string
NewName    string
```

`HostDevice` is for tap name

In GuestOS, if we want to add a new interface named `eth5`, and there are two existing network interface named `eth0` and `eth3`, when we insert a new tap device, the new added device will be named `eth1` in Guest, so we should use "eth1" as `DeviceName` which will be used by hyperstart to search for new added network interface.

then we user `eth5` as `NewName`, hyperstart will rename `eth1` to `eth5`. So in guest, we have 3 nic now: `eth0`, `eth3` and `eth5`, next time, newly added tap device will count from `eth1` again.

That's why I need to add a `NewName` field.

In above example, `HostDevice` is tap name, `DeviceName` is `eth1` and `NewName` is `eth5`. I can't remove `NewName` and just set `DeviceName=eth5`, we still need to know the `eth1` name.

That's why it's so complicated and confusing .